src/vsphere_cpi/spec/integration/client_spec.rb

```diff
@@ -38,105 +38,119 @@ module VSphereCloud
-          network = @client.find_network(@datacenter, ambiguous_portgroup)
-          expect(network).not_to be_nil
-          expected_name = ambiguous_portgroup.split('/').last
-          expect(network.name).to eq(expected_name)
-          expect(network).to be_a(VimSdk::Vim::Network)
+        it 'raises an ambiguous network error otherwise' do
+          expect do
+            @client.find_network(@datacenter, erroneous_dvpg_name)
+          end.to raise_error(/#{erroneous_dvpg_name}/)
         end
+      end
 
-        it 'returns the distributed virtual portgroup if it is prefixed with the distributed virtual switch\'s name' do
-          expected_network_name = distributed_virtual_portgroup.split('/')[-1]
-          expected_dvs_name = distributed_virtual_portgroup.split('/')[-2]
+      context 'if the network name is prefixed with a folder path' do
+        let(:standard_portgroup_path) { ENV.fetch('BOSH_VSPHERE_CPI_PORTGROUP_FOLDER_STANDARD') }
+        let(:dvpg_path) { ENV.fetch('BOSH_VSPHERE_CPI_PORTGROUP_FOLDER_DISTRIBUTED') }
 
-          network = @client.find_network(@datacenter, distributed_virtual_portgroup)
-          expect(network).not_to be_nil
-          expect(network.name).to eq(expected_network_name)
-          expect(network.config.distributed_virtual_switch.name).to eq(expected_dvs_name)
-          expect(network).to be_a(VimSdk::Vim::Dvs::DistributedVirtualPortgroup)
+        it 'returns nil if no such network exists' do
+          folder_name = standard_portgroup_path.split('/').first
+          network = @client.find_network(@datacenter, "#{folder_name}/absent-network")
+          expect(network).to be_nil
         end
 
-        it 'returns nil if the distributed virtual switch doesn\'t exist' do
-          folder_name = distributed_virtual_portgroup.split('/').first
-          network_name = distributed_virtual_portgroup.split('/').last
-          network = @client.find_network(@datacenter, "#{folder_name}/non-existent-dvs/#{network_name}")
+        it "returns nil if the DVS doesn't exist" do
+          folder_name, *_, name = dvpg_path.split('/')
+          network = @client.find_network(@datacenter, "#{folder_name}/absent-dvs/#{name}")
           expect(network).to be_nil
         end
 
-        it 'returns nil if the standard virtual portgroup doesn\'t exist' do
-          folder_name = standard_virtual_portgroup.split('/').first
-          network = @client.find_network(@datacenter, "#{folder_name}/non-existent-standard-portgroup")
-          expect(network).to be_nil
+        it 'returns the standard switch if it exists' do
+          network = @client.find_network(@datacenter, standard_portgroup_path)
+          expect(network).to be_an_instance_of(VimSdk::Vim::Network)
+          expect(network.name).to eq(standard_portgroup_path.split('/').last)
+        end
+
+        it 'returns the DVPG if it exists' do
+          network = @client.find_network(@datacenter, dvpg_path)
```

Comment:
This and test below test with same `dvpg_path`
```
dvpg_path is = BOSH_VSPHERE_CPI_PORTGROUP_FOLDER_DISTRIBUTED=vcpi-network-folder/vcpi-switch-2/vcpi-dvpg-2-1
```

I think this test should just use `vcpi-network-folder/vcpi-dvpg-2-1` (without DVS)